app/models/concerns/multi_searchable_helpers.rb

```diff
@@ -0,0 +1,71 @@
+module MultiSearchableHelpers
+  extend ActiveSupport::Concern
+
+  included do # rubocop:disable Metrics/BlockLength
+    PgSearch.multisearch_options = {
+      using: {
+        trigram: {
+          word_similarity: true,
+          threshold: 0.3
+        },
+        tsearch: { any_word: true }
+      }
+    }
+
+    def update_contact_search_document
+      return if contact_pg_search_record.present?
+
+      initialize_contact_pg_search_record.update!(
+        content: "#{contact.id} #{contact.email} #{contact.name} #{contact.phone_number} #{contact.account_id}",
+        conversation_id: id,
+        inbox_id: inbox_id
+      )
+    end
+
+    # NOTE: To add multi search records with contacts.name associated to conversation for previously added records.
+    # We can not find contacts.name from conversations directly so we added this joins here.
+    def self.rebuild_pg_search_documents(account_id)
+      return unless name == 'Conversation'
+
+      connection.execute <<~SQL.squish
+        INSERT INTO pg_search_documents (searchable_type, searchable_id, content, account_id, conversation_id, inbox_id, created_at, updated_at)
+          SELECT 'Conversation' AS searchable_type,
+                  conversations.id AS searchable_id,
+                  CONCAT_WS(' ', conversations.display_id, contacts.email, contacts.name, contacts.phone_number, conversations.account_id) AS content,
+                  conversations.account_id::int AS account_id,
+                  conversations.id::int AS conversation_id,
+                  conversations.inbox_id::int AS inbox_id,
+                  now() AS created_at,
+                  now() AS updated_at
+          FROM conversations
+          INNER JOIN contacts
+            ON conversations.contact_id = contacts.id
+          WHERE conversations.account_id = #{account_id}
+      SQL
+    end
+  end
```

Comment:
look and see if we can get rid of the SQL statements and use `record.update_pg_search_document ` instead. 
otherwise move the logic to migrations and make sure we escape values before running the sql.